src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -282,10 +284,7 @@ export function updateLayeredMaterialNodeImagery(context, layer, node) {
                     console.warn(`Imagery texture update error for ${node}: ${err}`);
                 }
                 const definitiveError = node.layerUpdateState[layer.id].errorCount > MAX_RETRY;
-                if (layer.protocol == 'wmts' || layer.protocol == 'wmtsc') {
-                    // keep the targetLevel that caused an error
-                    node.layerUpdateState[layer.id].targetLevel = targetLevel;
-                }
+                node.layerUpdateState[layer.id].failedFetchLevel = targetLevel;
```

Comment:
What about modifying `LayerUpdateState.failure` function to accept a optional argument,  that would be stored in a `failureParams` member.
Something like:

```js
node.layerUpdateState[layer.id].failure(Date.now(), definitiveError, targetLevel);
```

that could be used as:
```js
const failedFetchLevel = node.layerUpdateState[layer.id].failureParams.level;
```

WDYT?

Reply:
I updated the PR